dashboard/src/modules/components/ComparisonComponent/common-components.jsx

```diff
@@ -67,3 +70,35 @@ export const SearchByName = () => {
     />
   );
 };
+
+export const ViewOptions = (props) => {
+  const dispatch = useDispatch();
+  const [isOpen, setIsOpen] = React.useState(false);
+  const selected = useSelector((state) => state.comparison.datasetType);
+  const onSelect = (_event, value) => {
+    console.log("selected", value);
+    dispatch(setDatasetType(value, props.currPage));
```

Comment:
This construction is slightly disappointing:  `setDatasetType()` returns a thunk and, in order to use it, we have to fetch `dispatch()` at line 75; meanwhile, the only reason I see for using a thunk is that we didn't have `dispatch()` handy inside `setDatasetType()`, which is resolved by the fact that a thunk receives it as a parameter; however, since we've gone to the trouble of fetching it here, we _could_ just **pass** it into `setDatasetType()` as a parameter, so that the function can call it directly instead of returning a thunk (or, perhaps, `setDatasetType()` could have called `useDispatch()` itself...).

Reply:
Updated